src/main.ts

```diff
@@ -1,8 +1,17 @@
 import { NestFactory } from '@nestjs/core';
 import { AppModule } from './app.module';
+import {  ConfigService } from '@nestjs/config';
+import * as cookieParser from 'cookie-parser';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
-  await app.listen(3000);
+  app.enableCors({
+    exposedHeaders: ['x-auth-status']
+  })
+  app.use(cookieParser());
+  const configService = app.get(ConfigService);
+  const port = configService.get<number>('PORT') || 51004;
+  await app.listen(port);
+  console.log(`Server is running on http://localhost:${port}`);
 }
-bootstrap();
+bootstrap().then(() => {});
```

Comment:
If there is no .then(() => {}), it also can run successfully, but because I use the IntelliJ IDEA, it implement more serious type checking, so I add .then(() => {}) to avoid reminder.